test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopExeWtihNetStandardLib.cs

```diff
@@ -0,0 +1,334 @@
+        {
+        }
+
+        public enum ReferenceScenario
+        {
+            ProjectReference,
+            RawFileName,
+            HintPath
+        };
+
+        private void AddReferenceToLibrary(XDocument project, ReferenceScenario scenario)
+        {
+            var ns = project.Root.Name.Namespace;
+            var itemGroup = project.Root
+                .Elements(ns + "ItemGroup")
+                .Where(ig => ig.Elements(ns + "Reference").Any())
+                .FirstOrDefault();
+
+            if (itemGroup == null)
+            {
+                itemGroup = new XElement(ns + "ItemGroup");
+                project.Root.Add(itemGroup);
+            }
+
+            if (scenario == ReferenceScenario.ProjectReference)
+            {
+                itemGroup.Add(new XElement(ns + "ProjectReference",
+                    new XAttribute("Include", $@"..\{LibraryName}\{LibraryName}.csproj")));
+            }
+            else
+            {
+                var binaryPath = $@"..\{LibraryName}\bin\$(Configuration)\netstandard2.0\{LibraryName}.dll";
+                if (scenario == ReferenceScenario.HintPath)
+                {
+                    itemGroup.Add(new XElement(ns + "Reference",
+                        new XAttribute("Include", LibraryName),
+                        new XElement(ns + "HintPath", binaryPath)));
+                }
+                else if (scenario == ReferenceScenario.RawFileName)
+                {
+                    itemGroup.Add(new XElement(ns + "Reference",
+                        new XAttribute("Include", binaryPath)));
+                }
+            }
+        }
+
+        private string GetTemplateName(bool isSdk, bool usePackagesConfig = false)
+        {
+            return isSdk ? TemplateName : usePackagesConfig ? TemplateNamePackagesConfig : TemplateNameNonSdk;
+        }
+
+        private bool IsAppProject(string projectPath)
+        {
+            return Path.GetFileNameWithoutExtension(projectPath).Equals(AppName, StringComparison.OrdinalIgnoreCase);
+        }
+
+        private bool IsLibraryProject(string projectPath)
+        {
+            return Path.GetFileNameWithoutExtension(projectPath).Equals(LibraryName, StringComparison.OrdinalIgnoreCase);
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData(true, ReferenceScenario.ProjectReference)]
+        [InlineData(true, ReferenceScenario.RawFileName)]
+        [InlineData(true, ReferenceScenario.HintPath)]
+        [InlineData(false, ReferenceScenario.ProjectReference)]
+        [InlineData(false, ReferenceScenario.RawFileName)]
+        [InlineData(false, ReferenceScenario.HintPath)]
+        public void It_includes_netstandard(bool isSdk, ReferenceScenario scenario)
+        {
+            var testAsset = _testAssetsManager
+                .CopyTestAsset(GetTemplateName(isSdk), identifier: scenario.ToString())
```

Comment:
They use different templates which get included in the path.  I was actually hitting max-path at one point which is what I shortened the test method names and tried to be brief in these identifiers.